include/bout/sys/position.hxx

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "bout/region.hxx"
+
+#if defined(__cplusplus) && __cplusplus >= 201703L
+# define BOUT_FALL_THROUGH [[fallthrough]];
+#elif defined(__GNUC__) && __GNUC__ >= 7
+# define BOUT_FALL_THROUGH __attribute__ ((fallthrough));
+#else
+# define BOUT_FALL_THROUGH
+#endif
+
+
+class BoundaryRegion;
+class Mesh;
+
+class Position{
```

Comment:
`Position` is a super common class name, best stick this in a `bout` namespace -- and possibly one deeper, although I'm not sure what that should be yet